agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -3115,120 +2907,21 @@ bool slave_thread::slave_fsm(bool &call_slave_select)
-                               << config.hostap_iface;
-                    if (!set_wifi_iface_state(config.backhaul_wireless_iface,
-                                              WIFI_IFACE_OPER_RESTORE)) {
-                        LOG(ERROR)
-                            << "error changing backhaul wireless iface state --> slave_reset();";
-                        platform_notify_error(BPL_ERR_SLAVE_IFACE_CHANGE_STATE_FAILED,
-                                              config.backhaul_wireless_iface.c_str());
-                        stop_on_failure_attempts--;
-                        slave_reset();
-                        break;
-                    }
-                }
-            }
-
-            if (!platform_settings.local_gw) {
-                is_backhaul_manager   = false;
-                iface_status_bh_wired = eRadioStatus::OFF;
-            }
-
-            // mark slave as in non operational state
-            iface_status_operational_state = false;
-
-            LOG(TRACE) << "goto STATE_GET_WLAN_READY_STATUS";
-            slave_state = STATE_GET_WLAN_READY_STATUS;
-        }
-
-        break;
-    }
-    case STATE_GET_WLAN_READY_STATUS: {
-
-        auto request = message_com::create_vs_message<
-            beerocks_message::cACTION_PLATFORM_GET_WLAN_READY_STATUS_REQUEST>(cmdu_tx);
-
-        if (request == nullptr) {
-            LOG(ERROR) << "Failed building message!";
-            return false;
-        }
-
-        //LOG(DEBUG) << "Sending cACTION_PLATFORM_GET_WLAN_READY_STATUS_REQUEST";
-
-        if (!message_com::send_cmdu(platform_manager_socket, cmdu_tx)) {
-            LOG(ERROR) << "can't send message to platform manager!";
-            return false;
-        }
-
-        slave_state_timer =
-            std::chrono::steady_clock::now() +
-            std::chrono::seconds(STATE_WAIT_FOR_WLAN_READY_STATUS_RESPONSE_TIMEOUT_SEC);
-
-        //LOG(DEBUG) << "goto STATE_WAIT_FOR_WLAN_READY_STATUS_RESPONSE";
-        slave_state = STATE_WAIT_FOR_WLAN_READY_STATUS_RESPONSE;
-
-    } break;
-    case STATE_WAIT_FOR_WLAN_READY_STATUS_RESPONSE: {
-
-        //if timeout passed perform slave reset
-        if (std::chrono::steady_clock::now() > slave_state_timer) {
-            LOG(ERROR) << "STATE_WAIT_FOR_WLAN_READY_STATUS_RESPONSE timeout!";
-            platform_notify_error(BPL_ERR_SLAVE_TIMEOUT_GET_WLAN_READY_STATUS_REQUEST, "");
-            stop_on_failure_attempts--;
-            slave_reset();
-        }
-
-    } break;
-    case STATE_JOIN_INIT_BRING_UP_INTERFACES: {
-
-        if (!set_wifi_iface_state(config.hostap_iface, WIFI_IFACE_OPER_ENABLE)) {
-            LOG(ERROR) << "error changing iface state --> slave_reset();";
-            platform_notify_error(BPL_ERR_SLAVE_IFACE_CHANGE_STATE_FAILED,
-                                  config.hostap_iface.c_str());
-            stop_on_failure_attempts--;
-            slave_reset();
             break;
```

Comment:
Personally, I don't like the "break out of switch" pattern very much. It's OK-ish within a loop, but I find that breaking out of a switch makes the code flow harder to follow. So I'd have put (or rather, kept) an else here instead. (nitpick)

Reply:
I think otherwise, `break` statements help understand that you need to continue reading the code. Also, it can turn the code less nested.